### PR TITLE
Use stargate v1.0.1 and cleanup dockerfiles

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: cassandra-3.11/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+    && apt clean all \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/
 
-RUN unzip stargate-jars -x "*/persistence-*"
-RUN unzip stargate-jars "*/persistence-api*"
-RUN unzip stargate-jars "*/persistence-cassandra-3.11*"
+RUN curl -L https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip \
+    && unzip stargate-jars -x "*/persistence-*" \
+    && unzip stargate-jars "*/persistence-api*" \
+    && unzip stargate-jars "*/persistence-cassandra-3.11*"
```
Comment (dimas-b) [inline cassandra-3.11/Dockerfile:14]: do we want to remove the zip file at the end?

File: cassandra-3.11/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-ARG STARGATE_VERSION=v1.0.0
+ARG STARGATE_VERSION=v1.0.1
 
-RUN apt update && apt install curl unzip iproute2 libaio1 -y
-RUN curl -L https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip
+RUN apt update -qq \
+    && apt install curl unzip iproute2 libaio1 -y \
+    && apt autoremove --yes \
+    && apt clean all \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/
```
Comment (jimdickinson) [inline cassandra-3.11/Dockerfile:9]: I've seen issues with this line before, upstream images indicate this is safe

```suggestion
    && rm -rf /var/lib/apt/lists/*
```

File: dse-6.8/Dockerfile [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
-ARG STARGATE_VERSION=v1.0.0
+ARG STARGATE_VERSION=v1.0.1
 
-RUN apt update && apt install curl unzip iproute2 libaio1 -y
-RUN curl -sL https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip
+RUN apt update -qq \
+    && apt install curl unzip iproute2 libaio1 -y \
+    && apt autoremove --yes \
+    && apt clean all \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/
```
Comment (jimdickinson) [inline dse-6.8/Dockerfile:9]: here too

```suggestion
    && rm -rf /var/lib/apt/lists/*
```

**Review by jimdickinson — APPROVED** [review]
I forked and started this, and you beat me to the finish line 👍

File: cassandra-3.11/Dockerfile (hunk 1 above)
Comment (jimdickinson) [reply cassandra-3.11/Dockerfile:14]: good point - we should `rm` it

File: cassandra-4.0/Dockerfile [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
-ARG STARGATE_VERSION=v1.0.0
+ARG STARGATE_VERSION=v1.0.1
 
-RUN apt update && apt install curl unzip iproute2 libaio1 -y
-RUN curl -sL https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip
+RUN apt update -qq \
+    && apt install curl unzip iproute2 libaio1 -y \
+    && apt autoremove --yes \
+    && apt clean all \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/
```
Comment (dimas-b) [inline cassandra-4.0/Dockerfile:9]: Would it make sense to tag a local image at this point and reuse it for C* version-dependent images? Even if it does not help the size it should allow us to consolidate repeated `Dockerfile` commands and will probably reduce built time.

Comment (dougwettlaufer) [reply cassandra-4.0/Dockerfile]: done

**Review by dimas-b — APPROVED** [review]

**Review by jimdickinson — APPROVED** [review]

**Pull request merged**